src/IniFile.cpp

```diff
@@ -49,6 +49,9 @@ cIniFile::cIniFile(void) :
 
 bool cIniFile::ReadFile(const AString & a_FileName, bool a_AllowExampleRedirect)
 {
+
+	m_Filename = a_FileName;
+
```

Comment:
Because I need the cSettingsRepositoryInterface::Flush method to have the semantics of flush to backing storage, where backing storage may not be a file. Requiring this method to take a filename seriously messes up the semantics of cOverridesSettingsRepository::Flush().